src/setup-gcloud.ts

```diff
@@ -55,12 +55,20 @@ export async function run(): Promise<void> {
     }
 
     const serviceAccountKey = core.getInput('service_account_key');
-    // If a service account key isn't provided, log an un-authenticated notice
-    if (!serviceAccountKey) {
-      core.info('No credentials provided, skipping authentication');
-      return;
-    } else {
+    // If a service account key is provided, add warning to use google-github-actions/auth
+    if (serviceAccountKey) {
+      core.warning(
+        '"serviceAccountKey" input has been deprecated. ' +
+          'Please switch to using google-github-actions/auth which supports both Workload Identity Federation and JSON Key authentication. ' +
+          'For more details, see https://github.com/google-github-actions/setup-gcloud#authorization',
+      );
+    }
+
+    // Either serviceAccountKey or GOOGLE_GHA_CREDS_PATH env var required
+    if (serviceAccountKey || process.env.GOOGLE_GHA_CREDS_PATH) {
       await authenticateGcloudSDK(serviceAccountKey);
+    } else {
+      core.info('No credentials detected, skipping authentication');
```

Comment:
Maybe make this `debug`? I'm worried people will see this log message and think they did something incorrect

Reply:
Historically this [has been](https://github.com/google-github-actions/setup-gcloud/blob/94faef24d2af9aed96d8838c7252ae4f7165dc76/src/setup-gcloud.ts#L60) info lvl so folks should be used to it now. I also think it's easier for users to identify/self service issues like empty GH secrets if we log at info level as opposed to debug which requires setting ACTIONS_STEP_DEBUG.